### DISCUSSION TITLE
Handling Grant Rounds where the matching is not fully saturated

### DISCUSSION THREAD

**Discussion by phutchins** [dcgtc/dgrants#516] [Architecture & Design]
As we saw in the Moonshot Collective Grant Round, with the current implementation there will be times when a round will not reach full saturation and the funds will not be completely distributed locking the remaining funds on chain forever. Lets discuss how to handle this and the short, mid, and long term iterations that we can take to implement short term solutions then a more elegant solution in the future.

We discussed a few options but some of them have inherent problems:
1) Start out by assigning all of the remaining funds for each donation
- This creates a problem because in the very beginning of a round it would look like every grant getting donations is getting a HUGE amount of matching that quickly gets smaller creating a poor user experience
2) After the round ends, distribute the remaining funds based on quadratic funding in some way
- This seems like the best option currently
3) Have a clawback so that the entity running the grant round can retrieve any remaining funds
- This may be a good option for some configurations for instance where the funds were issued from a DAOs treasury. Those funds could then be returned to the treasury and reissued based on a vote from that DAO

I believe we should likely implement something like #2 initially then create an additional plugin option implementing #3.

I would love to hear any other ideas or options that anyone sees.

**Comment by ikwuoz** [comment]
I think the clawback mechanism fits a potentially short-term fix, and I am assuming the cause for this  is primarily *solidity* been unable to handle floating point arithmetics.

Also the numbering used in the above, point to existing PRs which might not be relevant in this context. You might want to fix that.
